app/src/main/java/org/desarrolladorslp/teckersapp/ui/deliverables/DeliverableAdapter.kt

```diff
@@ -0,0 +1,106 @@
+package org.desarrolladorslp.teckersapp.ui.deliverables
+
+import android.os.Build
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.annotation.RequiresApi
+import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.deliverable_item.view.*
+import org.desarrolladorslp.teckersapp.R
+import org.desarrolladorslp.teckersapp.model.DeliverableHeader
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+
+data class DeliverableAdapter(private val deliverablesHeader: ArrayList<DeliverableHeader>) :
+    RecyclerView.Adapter<DeliverableAdapter.DeliverableHeaderHolder>() {
+    private var deliverableCount =0
+
+    fun add(deliverableHeader: DeliverableHeader, position: Int) {
+        var position = position
+        position = if (position == -1) itemCount else position
+        deliverablesHeader.add(position, deliverableHeader)
+        notifyItemInserted(position)
+    }
+
+    fun remove(position: Int) {
+        if (position < itemCount) {
+            deliverablesHeader.removeAt(position)
+            notifyItemRemoved(position)
+        }
+
+    }
+
+    override fun getItemCount() = deliverablesHeader.size
+
+    @RequiresApi(Build.VERSION_CODES.O)
```

Comment:
What if the version code is below O?

Reply:
it is assign the O

Reply:
okey

Reply:
I am using code that it is accept on versions upper that the one i have on the minSDK

Reply:
Correct, what's the minSDK? Android O? We should check if this will be compatible with the average user's phones